solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -390,18 +391,19 @@ contract WalletRegistry is IRandomBeaconConsumer, Ownable {
     function approveDkgResult(EcdsaDkg.Result calldata dkgResult) external {
         uint32[] memory misbehavedMembers = dkg.approveResult(dkgResult);
 
-        bytes32 publicKeyHash = keccak256(dkgResult.groupPubKey);
-
-        wallets.addWallet(dkgResult.membersHash, publicKeyHash);
+        bytes32 walletID = wallets.addWallet(
+            dkgResult.membersHash,
+            dkgResult.groupPubKey
+        );
 
-        emit WalletCreated(publicKeyHash, keccak256(abi.encode(dkgResult)));
+        emit WalletCreated(walletID, keccak256(abi.encode(dkgResult)));
 
         // TODO: Disable rewards for misbehavedMembers.
         //slither-disable-next-line redundant-statements
         misbehavedMembers;
 
         walletOwner.__ecdsaWalletCreatedCallback(
-            publicKeyHash,
+            walletID,
```

Comment:
Would it make sense to return `(walletID bytes32, publicKeyX bytes32, publicKeyY bytes32)` from `wallets.addWallet` and pass `publicKeyX bytes32, publicKeyY bytes32` separately to `__ecdsaWalletCreatedCallback`?

I am looking at https://github.com/keep-network/tbtc-v2/pull/159 and I see we split `bytes` public key into x and y coordinate when compressing the key:
```
        bytes32 x = uncompressedPublicKey.slice32(0);
        bytes32 y = uncompressedPublicKey.slice32(3);
```

If `Bridge` needs separate coordinates, then maybe we can pass them to save gas on slicing?

Reply:
Yes, we can do that.

Reply:
... and I would leave `getWalletPublicKey` as-is because that's what we need for [`CheckBitcoinSigs` in `bitcoin-spv`](https://github.com/keep-network/bitcoin-spv/blob/solidity/v3.3.0-solc-0.8/solidity/contracts/CheckBitcoinSigs.sol#L20) that will be used by TBTC `Bridge` for fraud proofs.